src/Api/NoPersistent.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RankNTypes            #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE TemplateHaskell       #-}
+{-# LANGUAGE TypeOperators         #-}
+{-# LANGUAGE ViewPatterns          #-}
+
+{-# OPTIONS_GHC #-}
+
+module Api.NoPersistent
+where
```

Comment:
I suggest a rename this module to `Persistent` and create an abstract monad interface controlled via member visibility list, that Andres Löh mentioned. Even if we use this as prototyping, as it will help us to see how Persistent will fit into the big picture.